### PR TITLE
Update JSON_INHERITANCE.md

### LINKED ISSUE #28520: Document use of copy-from inheritance in different data types.
**Is your feature request related to a problem? Please describe.**  
Documentation for the json inheritance feature is sorely lacking, creating a great deal of confusion.  As far as I can tell the status quo is modders *think* that copy-from and related features are ubiquitous throughout the various json types.  In fact it is sporadically implemented across a subset of the json types, and sometimes even superceeded by a *different* inheritance implementation.
Additionally, the use of copy-from seems to be dominated by cargo-culting, copying what seems to work elsewhere instead of understanding how the feature works.
This also means that developers who are capable of doing something about the situation may not be aware of the lack of support.

To be clear, neither of these issues reflect poorly on modders, the problem is the lack of documentation.

**Describe the solution you'd like**  
Annotate entries in https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_INFO.md with details of integration with generic_factory.  There are three levels of integration:
``Supports copy-from`` means that modders can define entities of that type with a "copy-from" element and inherit from another entity of the same type.
``Supports extend`` means that in addition, modders can add an "extends" field to their definition to append entries to a list instead of overriding the entire list.
``Supports delete`` means moders can also add a "delete" field that removes elements from lists instead of overriding the entire list.

**Describe alternatives you've considered**  
Immediately migrating everything to generic_factory is worth doing, but is going to take a long time.

**Additional context**
These all seem to be related to not understanding copy-from or not knowing that it doesn't apply to a type of json entry.
https://github.com/CleverRaven/Cataclysm-DDA/issues/28206
https://github.com/CleverRaven/Cataclysm-DDA/issues/24702
https://github.com/CleverRaven/Cataclysm-DDA/issues/28508
https://github.com/CleverRaven/Cataclysm-DDA/issues/21557
https://github.com/CleverRaven/Cataclysm-DDA/pull/26666
https://github.com/CleverRaven/Cataclysm-DDA/issues/26275
https://github.com/CleverRaven/Cataclysm-DDA/issues/25384

### PR DESCRIPTION
**Pull request by RedMisao**
#### Summary
None

#### Purpose of change
Partially addresses, but does not cl0se #28520.

#### Describe the solution
Documentation of JSON inheritance is still very lacking, leading to confusion (monstergroups extending by default is news for a lot of people, #77278, not all JSON types that support `copy-from` are documented, fields related to json inheritance are completely missing, etc.)
Here, I try to address that by:
* Update the examples with current JSON syntax
* Adress the overall format (``` -> `, missing punctuation, double spaces, add JSON format to the examples, pseudo-lint, etc.)
* Move the monster example after the ammo examples, for consistency
* Added bullet points to the abstract section
* Edited explanations for clarity
* Added missing JSON types that support inheritance (and edit MATERIAL -> material)
* Add a Behavior section, indicating there's no default inheritance, with three "exception" examples I remember right now. I'm using that word here only because it's easier to understand. They're not true exceptions as there's no baseline (again, referenced issue).
  * Noteworthy: boolean `override` can be used to replace monstergroup lists. This is completely undocumented afaik and seemingly known only by word of mouth at the discord?
* Add mention of variants, and the specific case for guns, including links to the docs

#### Describe alternatives you've considered
The same PR with less changes.

#### Additional context
This **DOES NOT** cl0se #28520
This is a non-exhaustive update, regarding which `type`s are supported, and which are "exceptions", only what I remember seeing/reading/testing about

### CODE REVIEW

File: doc/JSON_INHERITANCE.md [hunk 1]
```diff
@@ -1,143 +1,195 @@
...
-The following additional rules apply to the above example:
 
-Chained inheritance is possible; for example ```reloaded_556``` inherits from ```556``` which is itself derived from ```223```
+* Chained inheritance is possible.  For example, `reloaded_556` inherits from `556`, which is itself derived from `223`.
+* Numeric values may be specified as `proportional` to the parent by via a decimal factor.  Here `reloaded_556` deals 90% of the damage of the factory equivalent with 110% dispersion.
+* Flags can be deleted via `delete`.  It is not an error if the deleted flag does not exist in the parent.
+
 
-Numeric values may be specified ```proportional``` to the parent by via a decimal factor where ```0.5``` is 50% and ```2.0``` is 200%.
+Not all `type`s work work the same.  For `"type": "MONSTER"`, `relative` would look slightly different and can be defined in two ways:
```
Comment (Night-Pryanik) [inline doc/JSON_INHERITANCE.md:78]:
```suggestion
Not all `type`s work the same.  For `"type": "MONSTER"`, `relative` would look slightly different and can be defined in two ways:
```

File: doc/JSON_INHERITANCE.md [hunk 2]
```diff
@@ -1,143 +1,195 @@
...
-  "proportional": { "melee_damage": 0.8 },
+    "//2": "first case, applies to cut damage",
+    "proportional": { "melee_damage": [ { "damage_type": "cut", "amount": 0.8 } ] },
+    "//3": "second case, applies to *all* melee damage",
+    "proportional": { "melee_damage": 0.8 },
 ```
 
-It is possible to define an ```abstract``` type that exists only for other types to inherit from and cannot itself be used in game. In the following condensed example ```magazine_belt``` provides values common to all implemented ammo belts:
+
+It is possible to define an `abstract` ID that exists only for other `type`s to inherit from and cannot itself be used in game.  This is done to facilitate maintanance and reduce line count.  In the following condensed example `magazine_belt` provides values common to all implemented ammo belts:
```
Comment (Night-Pryanik) [inline doc/JSON_INHERITANCE.md:103]:
```suggestion
It is possible to define an `abstract` ID that exists only for other `type`s to inherit from and cannot itself be used in game.  This is done to facilitate maintenance and reduce line count.  In the following condensed example `magazine_belt` provides values common to all implemented ammo belts:
```

**Comment by Zireael07** [comment]
Hint: reword "does not close " because the moment this is merged, GH will pick up on "close + issue number" and auto close which is NOT what you want

**Comment by Procyonae** [comment]
I did start on complete per type inheritance documentation several months ago but the amount of C++ digging and json testing it takes to check all the edge cases and weird combinations and then trying to format the results in a sensible way while including everything was alot

**Comment by RedMisao** [comment]
> Hint: reword "does not close " because the moment this is merged, GH will pick up on "close + issue number" and auto close which is NOT what you want

Done. Will it pass now?


> I did start on complete per type inheritance documentation several months ago but the amount of C++ digging and json testing it takes to check all the edge cases and weird combinations and then trying to format the results in a sensible way while including everything was alot

I believe it's better to just add known cases, so it's at least present _somewhere_ visible to everyone, as an attempt to prevent further confusion.
A real fix would be kevingranade's approach of migrating everything to generic_factory, or standardize the three cases of integration while specifying them in the docs.
Then again, I cannot do C++, so this is just janitor's work for a lack of baseline inheritance.

**Comment by Night-Pryanik** [comment]
> Done. Will it pass now?

In "does not close #number" GitHub reads only "close #number", i.e. it ignores the "does not" part. I fixed that for you by replacing "o" in "closes" with "0", I hope you don't mind.

**Pull request merged**